subsys/net/l2/ethernet/ethernet.c

```diff
@@ -498,15 +498,15 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	const struct ethernet_api *api = net_if_get_device(iface)->driver_api;
 	struct ethernet_context *ctx = net_if_l2_data(iface);
 	u16_t ptype;
-	int ret;
+	int err;
```

Comment:
See my comment in 2nd patch, ret needs to be set as pkt length on succes so this variable can return either error status or successful length sent. Keep it as ret.